### PR TITLE
Integration test using content-cache-backends-config charm

### PR DESCRIPTION
**Pull request by yhaliaw**
### Overview

After moving the `content-cache-backends-config` charm to this repo, the integration test can be done with the packing the `content-cache-backends-config` charm in the repo rather than download from CharmHub.

### Rationale

This allows testing of changes in `content-cache` and `content-cache-backends-config` charm together.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by alithethird — APPROVED** [review]

File: content-cache/docs/changelog.md [hunk 1]
```diff
@@ -1,21 +1,13 @@
...
+  * **Added**
+    * Support for TLS certificate integration. Integrating with charms that provides tls-certificates will enable HTTPS for the content cache.
+* 2024-10-17
+  * **Added**
+    * Support for fail-timeout, backends-path, proxy-cache-valid configuration options from integration with Content Cache Backends Config charm.
+* 2024-10-07
+  * **Added**
+    * Changelog for tracking user-relevant changes.
+    * Basic content cache functionality with nginx.
+    * Support for hostname, path, backends, protocol configuration options from integration with Content Cache Backends Config charm.
```
Comment (jdkandersson) [inline content-cache/docs/changelog.md:13]: The formatting doesn't look right?

File: content-cache/tests/integration/conftest.py [hunk 2]
```diff
@@ -47,37 +49,61 @@ def charm_file_fixture(pytestconfig: pytest.Config) -> str:
     return f"./{file}"
 
 
+@pytest_asyncio.fixture(name="config_charm_file", scope="module")
+async def config_charm_file_fixture(ops_test: OpsTest) -> AsyncIterator[str]:
```
Comment (jdkandersson) [inline content-cache/tests/integration/conftest.py:53]: Hmm, something is weird, this doesn't have a docstring, why doesn't our CI complain?

File: content-cache/docs/changelog.md (hunk 1 above)
Comment (yhaliaw) [reply content-cache/docs/changelog.md:13]: I moved it to list-based from heading-based due to some issue with review dog linting due to empty section after heading.

Let me try again, and see if operator-workflow still is using that linting.

File: content-cache/tests/integration/conftest.py (hunk 2 above)
Comment (yhaliaw) [reply content-cache/tests/integration/conftest.py:53]: I will look into this. Something might be wrong with the workflows.

**Comment by yhaliaw** [comment]
Converted back to draft due to some workflow-related issues.

File: content-cache/docs/changelog.md (hunk 1 above)
Comment (jdkandersson) [reply content-cache/docs/changelog.md:13]: You can also just add a short sentence or something

File: content-cache/tests/integration/conftest.py (hunk 2 above)
Comment (yhaliaw) [reply content-cache/tests/integration/conftest.py:53]: The issue is with the `pyproject.yaml` ignore some lints for files under `tests/*`.
I think `D100,D103` should not be ignored.
This is a issue with other repos as well such as https://github.com/canonical/github-runner-image-builder-operator/blob/704f3755d624e77b09af9e340a3fbebe7b5c7c32/pyproject.toml#L41.
I do know the scope of the issue across repos yet, but I think more repo are affect.

Comment (yhaliaw) [reply content-cache/tests/integration/conftest.py:53]: I updated the lint setting in pyproject and fix the docstrings.

**Review by jdkandersson — APPROVED** [review]
